scipy/ndimage/_measurements.py

```diff
@@ -307,6 +307,115 @@ def find_objects(input, max_label=0):
     return _nd_image.find_objects(input, max_label)
 
 
+def value_indices(arr, *, ignore_value=None):
+    """
+    Find indices of each distinct value in given array.
+
+    Parameters
+    ----------
+    arr : ndarray of ints
+        Array containing integer values.
+    ignore_value : int, optional
```

Comment:
Why not allow multiple ignore values?

Reply:
Good question. One of my earlier, private implementations of this did indeed accept a list of ignore_values, because I was trying to be as complete as I could. However, I have only ever used it with a single value. So, when I did this scipy version, I decided it was just adding unnecessary complexity. I would be happy to revisit this if there seemed to be a need, though.

Reply:
Is this argument worth it? Why not just drop the resulting key from the dictionary?

Reply:
I did think on this as well. I work in the context of earth observation satellite imagery. In such situations, it is common for there to be a significant NULL area around the real data, due to the way satellite ground track is not aligned with the usual map grid (and raster grid). I just checked a typical Landsat image, and the NULL area is 32% of the image. So, by not doing any counting for the NULL values, we save a bit of runtime, and more importantly a large piece of memory for the index values for every NULL pixel. So, it seemed worth doing, and does not seem to have made the code much uglier.